ElectricityBillApi.Tests/ElectricProviderPickerServiceTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿using ElectricityBillApi.Models;
+using ElectricityBillApi.Services;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace ElectricityBillApi.Tests
+{
+    public class ElectricProviderPickerServiceTests
+    {
+        [Fact]
+        public void FindCheapestProviderAndPrice_returnsNearCheapestProvider()
+        {
+            // arrange
+            var providers = GetTestData().ToList();
+            var iProviders = Substitute.For<IElectricProviders>();
+            iProviders.GetProviders.ReturnsForAnyArgs(providers);
+            var sut = new ElectricProviderPickerService(iProviders);
+            var ownAddress = new Address()
+            {
+                Location = new Location()
+                {
+                    X=0,
+                    Y=0,
+                    Z=0
+                }
+            };
+
+            // act 
+            var (cheapestProvider, calculatedPrice) = sut.FindCheapestProviderAndPrice(ownAddress);
+
+            // assert
+            Assert.Equal(providers.First(), cheapestProvider);
```

Comment:
If the order matters, don't use `IEnumerable`. Use an ordered list or an array or similar. When using `IEnumerable` the order of elements doesn't have to be kept.

Reply:
Changed it to a regular `List<ElectricProvider>`.